account_payment_term_discount/wizard/account_payment_register.py

```diff
@@ -39,88 +44,85 @@ def onchange_payment_amount(self):
+            payment_difference = self.payment_difference
+            self.payment_difference = 0.0
+            if payment_date <= till_discount_date:
+                discount_account = (
+                    payment_term.discount_income_account_id
+                    if self.invoice_id.is_purchase_document()
+                    else payment_term.discount_expense_account_id
+                )
+                # changing payment date
+                if not payment_difference and discount_amt:
+                    self.payment_difference = discount_amt
+                    self.payment_difference_handling = "reconcile"
+                    self.writeoff_account_id = discount_account.id
+                    self.writeoff_label = "Payment Discount"
+                # customer is paying more
+                elif payment_difference < discount_amt:
+                    if payment_difference > 0:
                         self.payment_difference = discount_amt
                         self.payment_difference_handling = "reconcile"
                         self.writeoff_account_id = discount_account.id
                         self.writeoff_label = "Payment Discount"
-                    # customer is paying more
-                    elif payment_difference < discount_amt:
-                        if payment_difference > 0:
-                            self.payment_difference = discount_amt
-                            self.payment_difference_handling = "reconcile"
-                            self.writeoff_account_id = discount_account.id
-                            self.writeoff_label = "Payment Discount"
-                        elif payment_difference < 0:
-                            self.payment_difference = payment_difference
-                    # ocustomer paying more than discount_amt
-                    elif payment_difference > discount_amt:
+                    elif payment_difference < 0:
                         self.payment_difference = payment_difference
-                        self.payment_difference_handling = "open"
-                        self.writeoff_label = False
-                    # customer paying more than discount_amt
-                    elif payment_difference == discount_amt and discount_amt > 0:
-                        self.payment_difference = abs(payment_difference)
-                        self.payment_difference_handling = "reconcile"
-                        self.writeoff_account_id = discount_account.id
-                        self.writeoff_label = "Payment Discount"
-                else:
+                # ocustomer paying more than discount_amt
+                elif payment_difference > discount_amt:
                     self.payment_difference = payment_difference
-
-                self.amount = amount_residual - self.payment_difference
-
-                self.amount_helper = self.amount
+                    self.payment_difference_handling = "open"
+                    self.writeoff_label = False
```

Comment:
I face this error in this case: tick on the option "Exclude Taxes from Discount?"
![image](https://github.com/user-attachments/assets/de9d04ce-b286-418d-9267-2bf772097465)